### ISSUE TITLE
SubnetSpecArgs doesn't apply tags

### ISSUE THREAD

**Issue by rodrigobrim** [pulumi/pulumi-awsx#963]
### What happened?

Subnets are created without the configured tags.

```
&ec2x.VpcArgs{
	CidrBlock: &cidrBlock,
	NumberOfAvailabilityZones: &numberOfAvailabilityZones,
	SubnetSpecs: []ec2x.SubnetSpecArgs{
		{
			Tags: pulumi.ToStringMap(map[string]string{
				"kubernetes.io/role/internal-elb": "1",
				"kubernetes.io/cluster/" + name:   "shared"}),
			Type:     ec2x.SubnetTypePrivate,
			CidrMask: &prvMask,
			Name:     &prvName,
		},
		{
			Tags: pulumi.ToStringMap(map[string]string{
				"kubernetes.io/role/elb":        "1",
				"kubernetes.io/cluster/" + name: "shared"}),
			Type:     ec2x.SubnetTypePublic,
			CidrMask: &pubMask,
			Name:     &pubName}}},
```

### Steps to reproduce

Using golang:

```
name := "my-network"

prvMask, pubMask := 23, 24
prvName, pubName := "private", "public"

vpc, error := ec2x.NewVpc(
ctx,
name,
&ec2x.VpcArgs{
	CidrBlock: pulumi.String("172.31.0.0/20")
	NumberOfAvailabilityZones: pulumi.Int(2),
	SubnetSpecs: []ec2x.SubnetSpecArgs{
		{
			Tags: pulumi.ToStringMap(map[string]string{
				"kubernetes.io/role/internal-elb": "1",
				"kubernetes.io/cluster/" + name:   "shared"}),
			Type:     ec2x.SubnetTypePrivate,
			CidrMask: &prvMask,
			Name:     &prvName,
		},
		{
			Tags: pulumi.ToStringMap(map[string]string{
				"kubernetes.io/role/elb":        "1",
				"kubernetes.io/cluster/" + name: "shared"}),
			Type:     ec2x.SubnetTypePublic,
			CidrMask: &pubMask,
			Name:     &pubName,
		},
	},
},
pulumi.Protect(true),
)
```

### Expected Behavior

Tags applied.

### Actual Behavior

Tags ignored.

### Output of `pulumi about`

```
CLI          
Version      3.48.0
Go Version   go1.19.2
Go Compiler  gc

Plugins
NAME        VERSION
aws         5.21.1
awsx        1.0.0
docker      3.2.0
eks         0.42.7
go          unknown
kubernetes  3.20.2

Host     
OS       ubuntu
Version  22.04
Arch     x86_64

This project is written in go: executable='/usr/local/go/bin/go' version='go version go1.18.2 linux/amd64'

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/rodrigo-brim-ctr
User           rodrigo-brim-ctr
Organizations  rodrigo-brim-ctr

Pulumi locates its logs in /tmp by default
omitted
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by roothorp** [comment]
Hi @rodrigobrim, thanks for the issue! I've raised this issue with the team so it can be prioritized and added to our work stack. I reproduced the issue in typescript too, not seeing the tags being added to a subnet.

**Comment by rodrigobrim** [comment]
Hi @roothorp! Thank you for the quick response.

I'd like to complement that to enable the [aws-load-balancer-controller](https://github.com/kubernetes-sigs/aws-load-balancer-controller/), I'm forced to create the network stack, add _IgnoreChanges([]string{"tags", "tagsAll"})_ _resourceOption_ on VPC, set the tags manually on AWS console, so then I can check the application deployment.

BTW, I'm mentioning it to describe/enforce the practical need for these tags.

**Comment by RobCannon** [comment]
This is critical for enabling the Application Load Balancer or Karpenter on EKS.  The subnets need specific tags and they are different for the public and private subnets.  Setting the tags attribute on the VpcArgs structure will not work properly since it will apply the same tags to public and private subnets.

**Comment by rodrigobrim** [comment]
Hey folks.

This bug is causing a lot of disorders. Is it possible to higher the priority?

**Comment by thomas-anderson-bsl** [comment]
I was running into this exact issue today, so I've submitted a PR to address this bug

**Comment by thomas-anderson-bsl** [comment]
The aforementioned PR has been merged, I would recommend we close this issue as it should be resolved now.

**Comment by RobCannon** [comment]
But there is not a release with this change.  The issue should not be closed until the fix is released.

**Comment by thomas-anderson-bsl** [comment]
This bug has been fixed and released with v1.0.2, recommend closing
